analyzers/tests/SonarAnalyzer.Test/TestCases/ClassShouldNotBeEmpty.cs

```diff
@@ -141,29 +141,58 @@ class Message { }                            // Compliant, ignored because of th
+    public class PublicClassWithInternalConstructor
+    {
+        internal PublicClassWithInternalConstructor() { }
+    }
 
-    class BaseWithInternalConstructor
+    public class PublicClassWithProtectedConstructor
     {
-        internal BaseWithInternalConstructor() { }
+        protected PublicClassWithProtectedConstructor() { }
     }
 
-    class WidensConstructorVisibility2 : BaseWithInternalConstructor { }            // Compliant
+    internal class InternalClassWithPublicConstructor
+    {
+        public InternalClassWithPublicConstructor() { }
+    }
 
-    class BaseWithPublicConstructor
+    internal class InternalClassWithInternalConstructor
     {
-        public BaseWithPublicConstructor() { }
+        internal InternalClassWithInternalConstructor() { }
     }
 
-    class ConstructorIsPublicAlready1 : BaseWithPublicConstructor { }               // Noncompliant
+    internal class InternalClassWithProtectedConstructor
+    {
+        protected InternalClassWithProtectedConstructor() { }
+    }
+
+    public class ConstructorIsPublicAlready1 : PublicClassWithPublicConstructor { }             // Noncompliant
+    public class WidensConstructorAccessibility1 : PublicClassWithInternalConstructor { }       // Compliant
+    public class WidensConstructorAccessibility2 : PublicClassWithProtectedConstructor { }      // Compliant
 
-    class BaseWithDefaultConstructor { }                                            // Noncompliant
+    internal class ConstructorIsPublicAlready2 : PublicClassWithPublicConstructor { }           // Noncompliant
+    internal class ConstructorIsInternalAlready1 : PublicClassWithInternalConstructor { }       // Noncompliant
+    internal class WidensConstructorAccessibility3 : PublicClassWithProtectedConstructor { }    // Compliant
+    internal class ConstructorIsInternalAlready2 : InternalClassWithPublicConstructor { }       // Noncompliant
+    internal class ConstructorIsInternalAlready3 : InternalClassWithInternalConstructor { }     // Noncompliant
+    internal class WidensConstructorAccessibility4 : InternalClassWithProtectedConstructor { }  // Compliant
 
-    class ConstructorIsPublicAlready2 : BaseWithDefaultConstructor { }              // Noncompliant
+    class ClassWithNestedClasses
+    {
+        private class PrivateClass1 : PublicClassWithPublicConstructor { }                      // Noncompliant
+        private class PrivateClass2 : PublicClassWithInternalConstructor { }                    // Noncompliant
+        private class PrivateClass3 : PublicClassWithProtectedConstructor { }                   // Noncompliant
+        private class PrivateClass4 : InternalClassWithPublicConstructor { }                    // Noncompliant
+        private class PrivateClass5 : InternalClassWithInternalConstructor { }                  // Noncompliant
+        private class PrivateClass6 : InternalClassWithProtectedConstructor { }                 // Noncompliant
```

Comment:
`private class PrivateClass6 : ProtectedClassWith###Constructor { }`
and 
`private class PrivateClass6 : PrivateClassWith###Constructor { }`